contracts/pkg/packageio/lock.go

```
@@ -0,0 +1,99 @@
+func canonicalAddress(root, address string) (string, error) {
+	if !packagecontract.IsLocalRuntimeAddress(address) {
+		return "", packagecontract.ErrInvalidFormat
+	}
+	if !strings.HasPrefix(address, "unix:") {
+		return address, nil
+	}
+	socketPath := strings.TrimPrefix(address, "unix:")
+	relative, err := filepath.Rel(root, socketPath)
+	if err != nil || relative == "." || relative == ".." || strings.HasPrefix(relative, ".."+string(filepath.Separator)) {
+		// 安装根外的绝对 Unix 地址是外部 executor 的稳定地址，不随安装根归一化。
+		return address, nil
```

golangci-lint (2.13.2)

[error] error is not nil (line 81) but it returns nil

(nilerr)